deploy/crd.yaml

```diff
@@ -11,9 +11,357 @@ spec:
+                        required:
+                          - name
+                        type: object
+                      type: array
+                    envFrom:
+                      items:
+                        properties:
+                          configMapRef:
+                            properties:
+                              name:
+                                type: string
+                              optional:
+                                type: boolean
+                            type: object
+                          prefix:
+                            type: string
+                          secretRef:
+                            properties:
+                              name:
+                                type: string
+                              optional:
+                                type: boolean
+                            type: object
+                        type: object
+                      type: array
+                resources:
+                  type: object
+                  properties:
+                    requests:
+                      type: object
+                      properties:
+                        memory:
+                          type: string
+                          pattern: ^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$
+                        cpu:
+                          type: string
+                          pattern: ^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$
+                        storage:
+                          type: string
+                          pattern: ^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$
+                        ephemeral-storage:
+                          type: string
+                          pattern: ^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$
+                    limits:
+                      type: object
+                      properties:
+                        memory:
+                          type: string
+                          pattern: ^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$
+                        cpu:
+                          type: string
+                          pattern: ^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$
+                        storage:
+                          type: string
+                          pattern: ^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$
+                        ephemeral-storage:
+                          type: string
+                          pattern: ^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$
+            taskManagerConfig:
+              type: object
+              properties:
+                taskSlots:
+                  type: integer
+                  minimum: 1
+                offHeapMemoryFraction:
+                  type: float
+                nodeSelector:
+                  type: object
+                  properties:
+                    additionalProperties:
+                      type: string
+                envConfig:
```

Comment:
Can we factor this out into its own object? That would avoid the need to duplicate it between TM/JM.

Reply:
I tried that I am unable to make components work for custom resources. Do you have any working example you can share ?

Reply:
Looks like it's not supported in kubernetes: https://github.com/kubernetes/kubernetes/issues/62872 😞